### ISSUE TITLE
Running mamba with v100 succeeded, but failed running mamba2

### ISSUE THREAD

**Issue by Li-muye** [MzeroMiko/VMamba#265]
Thanks for the support to Mamba2, but I met problems running classification task using mamba2 on v100 gpus.
my env:
- torch 1.12.1+cu116
- triton 2.1.0

config for mamba2 is copied from function vmamba_small_m2() in file vmamba.py:
`MODEL:
  TYPE: vssm
  NAME: mamba2_s
  DROP_PATH_RATE: 0.3
  NUM_CLASSES: 1000
  VSSM:
    EMBED_DIM: 96
    DEPTHS: [ 2, 2, 20, 2]
    PATCH_SIZE: 4
    IN_CHANS: 3
    SSM_D_STATE: 64
    SSM_DT_RANK: "auto"
    SSM_RATIO: 1.0
    SSM_ACT_LAYER: "gelu"
    SSM_CONV: 3
    SSM_CONV_BIAS: false
    SSM_DROP_RATE: 0.0
    SSM_INIT: "v2"
    SSM_FORWARDTYPE: "m0_noz"
    MLP_RATIO: 4.0
    MLP_DROP_RATE: 0.0
    GMLP: False
    PATCH_NORM: True
    DOWNSAMPLE: "v3"
    PATCHEMBED: "v2"
    NORM_LAYER: "ln"
    POSEMBED: False

TRAIN:
  USE_CHECKPOINT: False

DATA:
  IMG_SIZE: 224`

and the error log is:
`python: /project/lib/Analysis/Allocation.cpp:40: std::pair<llvm::SmallVector<unsigned int>, llvm::SmallVector<unsigned int> > mlir::triton::getCvtOrder(mlir::Attribute, mlir::Attribute): Assertion `!(srcMmaLayout && dstMmaLayout) && "Unexpected mma -> mma layout conversion"' failed.
WARNING:torch.distributed.elastic.multiprocessing.api:Sending process 22025 closing signal SIGTERM
/opt/conda/lib/python3.8/multiprocessing/resource_tracker.py:216: UserWarning: resource_tracker: There appear to be 32 leaked semaphore objects to clean up at shutdown
  warnings.warn('resource_tracker: There appear to be %d '`
  
I searched for the error and found others met the same problem using v100.
Is it mamba2's problem?

**Comment by MzeroMiko** [comment]
yes I think the problem is related to triton.

**Comment by xwmaxwma** [comment]
I also had this problem on v100, which can be avoided by using torch instead of triton. But another problem arises, after training several epochs, it reports error **loss is nan**. So how to solve it? Are you encountering this problem? I have not seen this problem when I use Mamba1.

**Comment by Li-muye** [comment]
I used the torch version of mamba2 and ran for 7 epochs on Tesla T4 GPU; everything is all right till now. I have not met the Nan loss. Did you set an improper learning rate?
